core/src/main/java/org/hibernate/ogm/dialect/eventstate/impl/EventContextManagerInitiator.java

```diff
@@ -30,7 +31,16 @@ private EventContextManagerInitiator() {
 	}
 
 	@Override
-	public EventContextManager initiateService(Map configurationValues, ServiceRegistryImplementor registry) {
-		return new EventContextManager();
+	public EventContextManager initiateService(SessionFactoryImplementor sessionFactory, SessionFactoryOptions sessionFactoryOptions,
+			ServiceRegistryImplementor registry) {
+		EventContextManager eventContextManager = new EventContextManager( registry );
+
+		GridDialect gridDialect = registry.getService( GridDialect.class );
+		// we don't use GridDialects.hasFacet here as we need to set it even for a {@link ForwardingGridDialect}
+		if ( gridDialect instanceof EventContextManagerAwareGridDialect ) {
```

Comment:
BatchOperationsDelegator is a ForwardingGridDialect and it needs the eventContextManager. See https://github.com/hibernate/hibernate-ogm/pull/711/files#diff-d064aa3d9da51990e6a0d80d519b9b6aR49 .

Same for InvocationCollectingGridDialect.